react/components/InputSearch/index.js

```diff
@@ -26,32 +38,62 @@ class InputSearch extends Component {
     this.props.onClear && this.props.onClear(event)
   }
 
+  handleSubmit = () => {
+    const { onSubmit, value } = this.props
+    onSubmit && onSubmit(value)
+  }
+
+  handleHovering = hover => {
+    this.setState({ hover })
+  }
+
+  handleFocus = focus => {
+    this.setState({ focus })
+  }
+
   render() {
-    const size = this.props.size
+    const { hover, focus } = this.state
+    const { size } = this.props
     const iconSize =
-      InputSearch.iconSizes[size] || InputSearch.iconSizes.default
+      InputSearch.iconSizes[size] || InputSearch.iconSizes.regular
 
     return (
-      <Input
-        {...this.props}
-        type="search"
-        suffix={
-          <span className="pointer">
-            {this.props.value ? (
-              <span
-                tabIndex={0}
-                onClick={this.handleClickClear}
-                className="pointer c-action-primary">
-                <DenyIcon size={iconSize} />
-              </span>
-            ) : (
-              <span className="c-link">
+      <div
+        onMouseEnter={() => this.handleHovering(true)}
```

Comment:
I think you want these hover callbacks to be in the child Input, not here, otherwise hovering the input label will trigger the hover effect.

Reply:
i don't know why they don't work in the child input, but where they are right now works with the expected behaviour.

Reply:
check this out http://g.recordit.co/qD5fhzP8Yw.gif